pkg/odo/cli/init/init.go

```diff
@@ -123,6 +124,13 @@ Your new component '%s' is ready in the current directory.
 To start editing your component, use 'odo dev' and open this folder in your favorite IDE.
 Changes will be directly reflected on the cluster.`, devfileObj.Data.GetMetadata().Name)
 
+	automateCommad := fmt.Sprintf("\nPort configuration using flag is currently not supported  \n\nYou can automate this command by executing:\n   odo init --name %s --devfile %s --devfile-registry %s", name, devfileLocation.Devfile, devfileLocation.DevfileRegistry)
+	if len(o.flags) == 0 && starterInfo != nil {
+		log.Infof("%s --starter %s", automateCommad, starterInfo.Name)
+	} else if len(o.flags) == 0 {
+		log.Infof("%s", automateCommad)
+	}
```

Comment:
I would separate the build of the comamnd string from the rest of the message.

Also, the condition of len(o.flags) could be placed n a better place.

I think you could instead write something like:

```
if len(o.flags) == 0 {
  automateCommad := fmt.Sprintf(odo init --name %s --devfile %s --devfile-registry %s", ...)
  if starterInfo != nil {
    automateCommand = fmt.Sprintf("%s --starter %s", automateCommad, starterInfo.Name
  }
  log.Infof("\nPort configuration using flag is currently not supported  \n\nYou can automate this command by executing:\n   %s\n", automateCommad)
}